File: app/src/main/java/com/dhsdevelopments/potato/clientapi/message/MessageElementCode.java
package com.dhsdevelopments.potato.clientapi.message;

import android.text.Spannable;
import android.text.SpannableString;
import com.dhsdevelopments.potato.CodeTypefaceSpan;

public class MessageElementCode extends TypedMessageElement
{
    public MessageElementCode( MessageElement content ) {
        super( content );
    }

    @Override
    public CharSequence getSpannable() {
        Spannable s = new SpannableString( content.getSpannable() );
        s.setSpan( new CodeTypefaceSpan(), 0, s.length(), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE );
        return s;
    }
}

File: app/src/main/java/com/dhsdevelopments/potato/clientapi/message/MessageElementCode.kt
package com.dhsdevelopments.potato.clientapi.message

import android.text.Spannable
import android.text.SpannableString
import com.dhsdevelopments.potato.CodeTypefaceSpan

class MessageElementCode(content: MessageElement) : TypedMessageElement(content) {
    override fun makeSpan(): CharSequence {
        val s = SpannableString(content.makeSpan())
        s.setSpan(CodeTypefaceSpan(), 0, s.length, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        return s
    }
}
